backup/manager.go

```diff
@@ -241,16 +243,102 @@ func (b *Manager) IsSafeToRunNode(nodeID string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	for _, s := range snapshots {
 		if s.NodeID == nodeID && s.BackupID != "" && backupID != s.BackupID {
 			b.log.Errorf("remote restore was found for node %v.", nodeID)
 			b.log.Errorf("current backupID=%v, remote backupID-%v", backupID, s.BackupID)
 			return false, nil
 		}
 	}
+
+	return true, nil
+}
+
+// LatestBackupInfo returns the latest backup time which is saved locally.
+// If the file does not exist, it returns the zero time and os.PathError.
+func (b *Manager) LatestBackupTime() (time.Time, error) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	file, err := os.Open(path.Join(b.workingDir, latestBackupSnapShot))
+	if err != nil {
+		b.log.Errorf("LatestBackupTime, detected err %v", err)
+		return time.Time{}, err
+	}
+	defer file.Close()
+	byteValue, err := io.ReadAll(file)
+	if err != nil {
+		b.log.Errorf("LatestBackupTime, detected err %v", err)
+		return time.Time{}, err
+	}
+	var snapshot SnapshotInfo
+	json.Unmarshal([]byte(byteValue), &snapshot)
+	return snapshot.ModifiedTime, nil
+}
+
+// EnsureLatestBackup checks whenever a given local state is newer than our snapshot.
+// if there is no local state, it returns true.
+func (b *Manager) EnsureLatestBackup(nodeID string) (bool, error) {
+	provider := b.GetProvider()
+	if provider == nil {
+		b.log.Infof("Provider is not set, EnsureLatestBackup return true")
+		return true, nil
+	}
+	snapshots, err := provider.AvailableSnapshots()
+	if err != nil {
+		return true, err
+	}
+	var snapshot SnapshotInfo
+	for _, s := range snapshots {
+		if s.NodeID == nodeID {
+			snapshot = s
+		}
+	}
+	if snapshot.ModifiedTime.IsZero() {
+		b.log.Infof("Time from snapshot is zero, EnsureLatestBackup returns true")
+		return true, nil
+	}
+	latestBackup, err := b.LatestBackupTime()
+	if err != nil {
```

Comment:
On new install the latest backup time won't find a file. Will it return error?
If so then a new user will get a conflict error.

Reply:
True, in this case do we wish to call  `SetLatestBackupTime` and return **true**?  
This way we avoid calling `AvailiableSnapshot` too often.

Reply:
Actually I think I am wrong here as this function will return earlier with empty snapshots in the cloud.
So if there is a snapshot in the cloud and the LatestBackupTime doesn't find a file we should show backup conflict.
What do you think?

Reply:
Yes, if the snapshot in the cloud is empty this will return early. 

Ideally I think we can return `false` here since we call [`UpdateLatestBackupTime`](https://github.com/breez/breez/blob/10ba088da5ae5b3e22bdb3040c1588f571821f3e/bindings/api.go#L341) on restore and on [successful backup request ](https://github.com/breez/breez/blob/10ba088da5ae5b3e22bdb3040c1588f571821f3e/backup/manager.go#L387).

On the other hand an issue I can think of is a race condition when calling `EnsureLatestBackup`  in `watchDaemonEvents` and on the first uploading of a backup in the `start` method for the backup manager. If the user triggers the first backup we risk that we call `ensureLatestBackup` right after a snapshot is available, but right before we manage to create the file for it.  

Not sure what the wished behaviour might be or how likely this scenario is, but a solution I can think of is calling `UpdateLatestBackupTime` when starting the backupmanager outside of the go function. And this way we can be sure that  if the file does not exist we may return `false` and prompt the user with an error. The downside of this is yet another call to `AvailableSnapshots`.

Reply:
Should be fixed in latest commit.